admin_bans.php

```diff
@@ -76,6 +76,10 @@
 				$result = $db->query('SELECT registration_ip FROM '.$db->prefix.'users WHERE id='.$user_id) or error('Unable to fetch user info', __FILE__, __LINE__, $db->error());
 				$ban_ip = ($db->num_rows($result)) ? $db->result($result) : '';
 			}
+
+			// Disable localhost IP ban
+			if (in_array($ban_ip, array('', '127.0.0.1', '::1'), true))
+				$ban_ip = '';

```

Comment:
Oh, never mind that first remark, I just noticed the `($ban_ip != '')` later in the code. The string check still does not make a lot of sense though.